src/box/box.cc

```diff
@@ -3473,12 +3478,41 @@ bootstrap_from_master(struct replica *master)
 	enum applier_state wait_state = replication_anon ?
 					APPLIER_FETCH_SNAPSHOT :
 					APPLIER_INITIAL_JOIN;
-	applier_resume_to_state(applier, wait_state, TIMEOUT_INFINITY);
+	try {
+		applier_resume_to_state(applier, wait_state,
+					TIMEOUT_INFINITY);
+	} catch (...) {
+		return -1;
+	}
+
+	try {
+		applier_resume_to_state(applier, APPLIER_CONTINUE_SNAPSHOT,
+					TIMEOUT_INFINITY);
+	} catch (...) {
+		return -1;
+	}
+
+	return 0;
+}
+
+/**
+ * Bootstrap from the remote master
+ * \pre master->applier->state == APPLIER_READY
```

Comment:
Not true really. You rely on it being in `APPLIER_CONTINUE_SNAPSHOT`, don't you?

Reply:
Fixed